app/controllers/topics_controller.rb

```diff
@@ -16,6 +16,14 @@ def create
     end
   end
 
+  def show
+    @topic = Topic.find_by_id(params[:id])
```

Comment:
`find_by_id` is a dynamic finder method that relies on `method_missing` to construct the appropriate SQL query on the fly. This makes dynamic finders not as performant as a standard method.

In general you should use the `find_by` method instead of dynamic finders
```
Topic.find_by(my_attributes: params[:my_attribute])
```

However, in this case the basic find (which searches by id) does the job, so let's just go with that:
```
Topic.find(params[:id])
```